gitpod-io/gitpod
# [content-service] add prestop hook to extract git status

## Description

add prestop hook to extract git status when workspace is using pvc.
This is needed when workspace is using PVC and ws-daemon cannot access those files directly because:
they are on a dedicated pvc
finalize happens when container is stopped, and at that point pvc has been detached.

So we use prestop hook to do necessary work right before container will receive signal to stop (later prebuilds will extract build log in the same way for example).

## Related Issue(s)

Part of https://github.com/gitpod-io/gitpod/pull/9475

## How to test

You can test this as part of https://github.com/gitpod-io/gitpod/pull/9475
But for this PR, verify that workspace starts and stops normally. This PR shouldn't affect normal workflow (without PVC) and that is what should be tested in this PR.

## Release Notes

```release-note
none
```

## Documentation
